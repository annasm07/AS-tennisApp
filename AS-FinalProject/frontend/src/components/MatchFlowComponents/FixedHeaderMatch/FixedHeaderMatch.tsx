import * as React from 'react';
import {StyleSheet, Image, TouchableOpacity} from 'react-native';
import {SafeAreaView} from 'react-native-safe-area-context';

export default function FixedHeaderMatch() {
  return (
    <SafeAreaView style={styles.header}>
      <TouchableOpacity>
        <Image
          style={styles.back}
          source={require('../../../images/arrow-back.png')}
        />
      </TouchableOpacity>
      <Image
        style={styles.icon}
        source={require('../../../images/ball-logo.png')}
      />
      <Image
        style={styles.iconRight}
        source={require('../../../images/three-dots.png')}
      />
    </SafeAreaView>
  );
}

const styles = StyleSheet.create({
  header: {
    borderBottomColor: 'gray',
    borderBottomWidth: 1,
    marginHorizontal: 10,
    flexDirection: 'row',
    marginBottom: -40,
  },
  back: {
    width: 30,
    height: 30,
    alignSelf: 'flex-start',
    marginBottom: -30,
    marginTop: 10,
    marginLeft: 20,
  },
  icon: {
    width: 50,
    height: 50,
    marginLeft: 102,
    marginBottom: -30,
    left: '40%',
  },
  iconRight: {
    marginTop: 14,
    marginLeft: 135,
    marginBottom: -30,
    left: '40%',
  },
});
