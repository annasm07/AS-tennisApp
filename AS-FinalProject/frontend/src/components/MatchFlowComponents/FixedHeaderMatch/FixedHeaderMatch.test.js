import {render} from '@testing-library/react-native';
import React from 'react';
import ConfigureStore from 'redux-mock-store';
import {Provider} from 'react-redux';
import FixedHeaderMatch from './FixedHeaderMatch';

describe('When invoked a FixedHeaderMatch component', () => {
  const mockStore = ConfigureStore();
  const store = mockStore({
    user: {user: {playerId: '123cde'}},
  });
  test('Should render FixedHeaderMatch component', () => {
    const fixedHeaderMatch = render(
      <Provider store={store}>
        <FixedHeaderMatch />
      </Provider>,
    );
    expect(fixedHeaderMatch).toMatchSnapshot();
  });
});
