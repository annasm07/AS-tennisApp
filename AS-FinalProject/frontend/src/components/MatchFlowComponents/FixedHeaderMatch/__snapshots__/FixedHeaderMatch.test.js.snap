// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`When invoked a FixedHeaderMatch component Should render FixedHeaderMatch component 1`] = `
<RNCSafeAreaView
  style={
    Object {
      "borderBottomColor": "gray",
      "borderBottomWidth": 1,
      "flexDirection": "row",
      "marginBottom": -40,
      "marginHorizontal": 10,
    }
  }
>
  <View
    accessible={true}
    focusable={false}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
  >
    <Image
      source={
        Object {
          "testUri": "../../../src/images/arrow-back.png",
        }
      }
      style={
        Object {
          "alignSelf": "flex-start",
          "height": 30,
          "marginBottom": -30,
          "marginLeft": 20,
          "marginTop": 10,
          "width": 30,
        }
      }
    />
  </View>
  <Image
    source={
      Object {
        "testUri": "../../../src/images/ball-logo.png",
      }
    }
    style={
      Object {
        "height": 50,
        "left": "40%",
        "marginBottom": -30,
        "marginLeft": 102,
        "width": 50,
      }
    }
  />
  <Image
    source={
      Object {
        "testUri": "../../../src/images/three-dots.png",
      }
    }
    style={
      Object {
        "left": "40%",
        "marginBottom": -30,
        "marginLeft": 135,
        "marginTop": 14,
      }
    }
  />
</RNCSafeAreaView>
`;
